### ISSUE TITLE
[a11y]: Accessibility issue with InlineLoading component

### ISSUE THREAD

**Issue by davidicus** [carbon-design-system/carbon#13464]
### Package

carbon-components-react

### Browser

N/A

### Operating System

MacOS, Other

### Package version

v7.56.0

### React version

16.14.0+

### Automated testing tool and ruleset

IBM Equal Access Accessibility-Checker - latest deployment

### Description

The `accessibility-checker` tool has recently deployed a [fix](https://github.com/IBMa/equal-access/pull/1329) to their existing ruleset. This involves all components that use a div (or any element with `role=generic`) with certain other aria attributes:

`aria-brailleroledescription`
`aria-label`
`aria-labelledby`
`aria-roledescription` 

Our failures involve the `InlineLoading` component but other IBMers in the thread mentioned the Carbon Modal as well. 

### WCAG 2.1 Violation

Fail_invalid_role_attr

### Reproduction/example

This error will throw if you run your accessibility scans using the latest version of the ruleset for "IBM Accessibility"

### Steps to reproduce

Update your `ruleArchive` entry in the [config](https://github.com/carbon-design-system/carbon/blob/b955ebe4a54f7cc10619f95c5bca1a3c64eb6344/achecker.js#L13) file to the latest entry. Once you run your test you will see the error. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by keithcha** [comment]
+1 for this issue.  For me, I get

`The ARIA attributes "aria-labelledby" are not valid for the element <div> with implicit ARIA role "generic"` that refers to a Modal - `<div aria-labelledby="bx--modal-header__heading--modal-2" class="bx--modal-content" id="bx--modal-body--modal-2"`

**Comment by tw15egan** [comment]
Thanks for reporting, we'll take a look. Any idea why these violations don't appear for me when using the Chrome Dev Tool version of the a11y checker with the latest deployment?

**Comment by davidicus** [comment]
That I do not know. Do you know if there is a role that get automatically assigned to this div by the browser? The difference between the extension and the node package would be the browser environment.

**Comment by davidicus** [comment]
I just spoke with the dev lead for the Accessibility tools. It looks like it may not have the latest update. 

>The extension lags a little in the rule updates. Chrome no longer lets us pull rules from the server, so they have to be packed with the extension. I’m not sure that the extension update has happened yet.

**Comment by ljcarot** [comment]
@tombrunet @mbgower FYI

**Comment by QxiaoQ** [comment]
This one can be reproduced in official story book.
<img width="1558" alt="image" src="https://user-images.githubusercontent.com/31646170/231101536-bd686850-c48d-44b4-9e16-8e572e540a2e.png">

**Comment by tombrunet** [comment]
For some background...  elements that have no implicit or explicit role (i.e., div or span without a role attribute) are internally given a role of `generic`. Per ARIA, `generic` elements are prohibited from having names (labels):
https://www.w3.org/TR/wai-aria-1.2/#generic

**Comment by keithcha** [comment]
What's the outlook on this issue?  Is there a workaround?

**Comment by mbgower** [comment]
Speaking directly to the inline loading, if this continues to be a problem, I would suggest giving the div a` role="alert"`
This has the exact same affect as setting the `aria-live="assertive"` yet gives the div a role, thus not failing this rule.

@tombrunet any reason this wouldn't be an approach for any of these inline loading messages?

**Comment by keithcha** [comment]
> What's the outlook on this issue? Is there a workaround?

This issue is being flagged as a failure with our accessibility compliance.  Without a fix, we would need to file a deviation.  Is there room to have this issue addressed sooner than later?

**Comment by tombrunet** [comment]
@keithcha Which ruleset and Carbon version are you running? I do not see the error on these storybooks with the Aug 9 ruleset:
https://react.carbondesignsystem.com/iframe.html?args=&id=components-inlineloading--default&viewMode=story
https://react.carbondesignsystem.com/iframe.html?args=&id=components-modal--default&viewMode=story

**Comment by keithcha** [comment]
@tombrunet You're right, the error went away.  I did some digging and found that #13742 fixed it.  I'm currently on carbon-components-react 7.56.0.  I'm not sure which version of carbon-components-react I should upgrade to in order to get the fix.  Based on the last comment in #13840 (which references #13742), perhaps the fix is in v11 and I need it backported to v10 in order to get the fix?

**Comment by keithcha** [comment]
> > @tombrunet You're right, the error went away. I did some digging and found that #13742 fixed it. I'm currently on carbon-components-react 7.56.0. I'm not sure which version of carbon-components-react I should upgrade to in order to get the fix. Based on the last comment in #13840 (which references #13742), perhaps the fix is in v11 and I need it backported to v10 in order to get the fix?
> 
> I stumbled on #14170 that backports #13742 to v10. I'll give the latest v10 carbon-components-react build a try and report back.

The same error but with a modal is fixed and available in both V11 and V10.

**Comment by tw15egan** [comment]
Not seeing any a11y violations in `InlineLoading` or `Modal` in both V11 and V10.
